src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -236,6 +236,23 @@ private void ValidateLinuxSKUConfiguration()
             {
                 _logger.LogError($"Unable to load the functions payload since the app was not provisioned with valid {AzureWebJobsSecretStorage} connection string.");
             }
+
+            var functionsTimeZone = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsTimeZone);
+            var functionsWebsiteTimeZone = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsWebsiteTimeZone);
+
+            // If we have a linux consumption app and the time zone env variable is set, log a warning and diagnostic event
+            if (EnvironmentExtensions.IsLinuxConsumptionOnAtlas(_environment) &&
```

Comment:
Would it work for flex consumption?

Reply:
I believe so. The issue is only with linux consumpiton apps. @brettsam is that correct?

Reply:
Linux consumption apps can run either on Atlas or Legion(Flex consumption) platform.

There is a `IsLinuxConsumptionOnLegion` method which also you may also need to use.

https://github.com/Azure/azure-functions-host/blob/f4fa1b1a86586c0964d5f38776654f9ea2e07abd/src/WebJobs.Script/Environment/EnvironmentExtensions.cs#L319-L325

Reply:
I wouldn't worry about Legion on this -- I *think* that these settings are meant to work there. Let's just be pragmatic here and get this in for the one we know is an issue -- and if we need to come back to it and improve it for Legion, we can do that later.